components/autocomplete/index.js

```diff
@@ -169,48 +329,87 @@ class Autocomplete extends Component {
 				this.select( options[ selectedIndex ] );
 				break;
 
+			case LEFT:
+			case RIGHT:
+			case TAB:
+				this.reset();
+				return;
+
 			default:
 				return;
 		}
 
 		// Any handled keycode should prevent original behavior. This relies on
 		// the early return in the default case.
 		event.preventDefault();
-		event.stopImmediatePropagation();
+		event.stopPropagation();
+	}
+
+	toggleKeyEvents( isListening ) {
+		// This exists because we must capture ENTER key presses before Editable.
+		// It seems that react fires the simulated capturing events after the
+		// native browser event has already bubbled so we can't stopPropagation
+		//  and avoid Editable getting the event from TinyMCE, hence we must
+		// register a native event handler.
+		if ( ! this.node ) {
```

Comment:
When is `this.node` falsey?

Reply:
This won't be called very often so I doubt it will have much effect on performance but I have removed the guard as requested.

Reply:
It's less about performance or safe-guarding, more to having knowledge to when those safe-guards should be expected to be encountered.